### PR TITLE
Implement REST API above CncReader

### PR DESCRIPTION
**Pull request by daabramoff**

### CODE REVIEW

**Comment by daabramoff** [comment]
https://github.com/easy-logic/aeron-monitoring/issues/17

File: web/src/main/java/io/aeron/monitoring/SwaggerConfig.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                .contact(new Contact("",
+                        "https://github.com/easy-logic/aeron-monitoring",
+                        ""))
+                .version("0.0.1")
+                .termsOfServiceUrl("")
+                .build();
+    }
+
+    @Override
+    protected void addResourceHandlers(ResourceHandlerRegistry registry) {
```
Comment (QIvan) [inline web/src/main/java/io/aeron/monitoring/SwaggerConfig.java:42]: final in parameters =(

File: web/src/main/java/io/aeron/monitoring/api/CncSnapshotController.java [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        return counters.get(counter);
+    }
+
+    private ChannelInfo getCncChannelInfo(String channel) {
+        CncSnapshot snapshot = cncReader.read();
+        Map<String, ChannelInfo> counters = snapshot.getChannels();
+        return counters.get(channel);
+    }
+
+    @RequestMapping("{mediaDriver}/version")
```
Comment (QIvan) [inline web/src/main/java/io/aeron/monitoring/api/CncSnapshotController.java]: as discussed `mediaDriver` should be a RequestParam

File: web/src/main/java/io/aeron/monitoring/api/CncSnapshotController.java [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+import java.util.Map;
+
+@RestController
+@RequestMapping(value = "/api/v1/cnc", method = RequestMethod.GET)
+@Api
+public class CncSnapshotController {
+    private CncReader cncReader = new CncReader();
+
+    @Value("${aeron.mediaDriver.baseDir:/dev/shm}")
+    private String baseDir;
```
Comment (QIvan) [inline web/src/main/java/io/aeron/monitoring/api/CncSnapshotController.java]: as discussed 
should be 
```
- "mediaDriverName" : "/path/to/media/driver"
```

File: web/src/main/java/io/aeron/monitoring/api/CncSnapshotController.java [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+                                    @ApiParam("Media Driver directory to be read")
+                                    String mediaDriver,
+                                    @PathVariable("channel")
+                                    @ApiParam("URL which specifies channel")
+                                    String channel,
+                                    @PathVariable("streamId")
+                                    @ApiParam("Stream identifier")
+                                    Integer streamId) {
+        ChannelInfo channelInfo = getCncChannelInfo(channel);
+        Map<Integer, StreamInfo> streams = channelInfo.getStreams();
```
Comment (QIvan) [inline web/src/main/java/io/aeron/monitoring/api/CncSnapshotController.java]: final for variables too =(

File: web/src/main/java/io/aeron/monitoring/api/CncSnapshotController.java [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+
+    @RequestMapping("{mediaDriver}/channels/{channel}")
+    @ApiOperation("Return information about single channel specified by parameter")
+    public ChannelInfo getChannelInfo(@PathVariable("mediaDriver")
+                                      @ApiParam("Media Driver directory to be read")
+                                      String mediaDriver,
+                                      @PathVariable("channel")
+                                      @ApiParam("URL which specifies channel")
+                                      String channel) {
+        ChannelInfo channelInfo = getCncChannelInfo(mediaDriver, channel);
```
Comment (QIvan) [inline web/src/main/java/io/aeron/monitoring/api/CncSnapshotController.java]: inline unnecessary variables

**Review by QIvan — COMMENTED** [review]
looks good! =)

**Comment by daabramoff** [comment]
@QIvan 
TBD later in separate tasks:
- configure the application with the the list of media driver directories
- create data transfer objects layer or slightly redesign internal API in order to return List instead of Map

**Comment by QIvan** [comment]
@daabramoff ok.
create these tasks please asap
thanks

**Pull request merged**